azurerm/internal/services/keyvault/key_vault_key_encrypt_resource.go

```diff
@@ -0,0 +1,94 @@
+package keyvault
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/2016-10-01/keyvault"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmKeyVaultKeyEncrypt() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKeyVaultKeyEncryptCreate,
+		Read:   schema.Noop,
+		Delete: schema.Noop,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			return fmt.Errorf("resource `azurerm_key_vault_key_encrypt` does not support `import` operation")
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			// avoid test fail, add timeout config for read and delete
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
```

Comment:
does it make sense to add a name property so we can identify this and use it in the ID?